.github/workflows/clangtidy_review.yml

```diff
@@ -0,0 +1,35 @@
+name: Clang tidy Review
+
+on: [pull_request]
+
+jobs:
+  check:
+    name: Clang tidy Review 
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+        
+      - name: "Install dependencies"
+        run: |
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+          sudo apt-get update
+          sudo apt-add-repository "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-10 main"
+          sudo apt-get install -y clang-10 clang-tidy-10 libboost-all-dev
```

Comment:
here you are install libboost from the ubuntu 18.04 packages.
is that wanted? we should clone the repo and test either the development or master branch

Reply:
The bare basic minimum for `clang-tidy` is that it finds all includes in codes it is running on, Since we use `boost/config` and other boost libs, it is required that clang-tidy finds all these includes.

Now, there are two ways for it:
 - Setup Boost ROOT or
 - Use `libboost-all-dev` so that all Boost headers are found in the system includes.

We then run the clang-tidy checks with `-Iinclude` flag, this makes sure that we are not testing on system boost.ublas rather in our repository. 

This is quicker than setting up boost root since we don't care about clang-tidy warnings from other Boost libraries, we can use any boost version. As for the build test, We are running with boostorg/boost from the master.

Reply:
Does it really matter what boost library we use, as long as we are able to lint our repository with any boost version. In Builds and Sanitizer checks, We do use Boost from master.

If you want, I can setup boost from master as well. But that will increase time for checks as everytime it will need to compile and install Boost from source.

Reply:
> Does it really matter what boost library we use, as long as we are able to lint our repository with any boost version. In Builds and Sanitizer checks, We do use Boost from master.

Generally, I am for your approach. Less work! However, the boost-package in ubuntu-20.04 could be version1.73 while the master on github could be already 1.75 and all of a sudden we get a compile error.

Reply:
Okay, I will build boost from source master and compile and install it. I will check if it's possible to cache it as well, so we can speed up things a bit.

Reply:
Best way would be to pull latest boost libraries as tarbar from web and install them. What do you say? It will be faster than compiling master from source, and latest as well.

Reply:
I am using 20.04 with `boostorg/master`. Since we only need headers files. I clone master and clone all submodules (shallow) then generate headers with `b2 headers` and manually copy them to `/usr/include`. while running clang-tidy.